codexctl/device.py

```diff
@@ -605,6 +780,86 @@ def install_sw_update(self, version_file: str) -> None:
             print("Update complete and device rebooting")
             os.system("reboot")
 
+    def _update_paper_pro_bootloader(self, bootloader_script: bytes, imx_boot: bytes) -> None:
+        """
+        Update bootloader on Paper Pro device for 3.22+ -> <3.22 downgrades.
+
+        This method uploads the bootloader script and image to the device,
+        then runs the update script twice (preinst and postinst) to update
+        both boot partitions.
+
+        Args:
+            bootloader_script: Contents of update-bootloader.sh
+            imx_boot: Contents of imx-boot image file
+
+        Raises:
+            SystemError: If bootloader update fails
+        """
+        import tempfile

```

Comment:
This should be at the top of the file. Imports should only be inline to deal with cyclical import issues, or if they are optional dependencies that need to be feature detected only if the call is made.